packages/nimble-components/src/wafer-map/modules/rendering.ts

```diff
@@ -49,61 +64,70 @@ export class RenderingModule {
         return '';
     }
 
-    private sortDies(): void {
-        this.dies = this.wafermap.dataManager.diesRenderInfo.sort((a, b) => {
-            if (a.fillStyle > b.fillStyle) {
-                return 1;
-            }
-            if (b.fillStyle > a.fillStyle) {
-                return -1;
-            }
-
-            return 0;
-        });
-    }
-
     private renderDies(): void {
-        let prev: DieRenderInfo | undefined;
+        let fillStyle = '';
+        const context = this.wafermap.canvasContext;
+        let [trCanvasMinPointX, trCanvasMinPointY] = this.wafermap.transform.invert([0, 0]);
+        const [trCanvasMaxPointX, trCanvasMaxPointY] = this.wafermap.transform.invert([
+            this.wafermap.canvas.width,
+            this.wafermap.canvas.height
+        ]);
+        const dieWidth = this.wafermap.dataManager.dieDimensions.width;
+        const dieHeight = this.wafermap.dataManager.dieDimensions.height;
+        trCanvasMinPointX -= dieWidth;
+        trCanvasMinPointY -= dieHeight;
 
         for (const die of this.dies) {
-            if (!prev) {
-                this.wafermap.canvasContext.fillStyle = die.fillStyle;
+            if (

```

Comment:
I think that the content of this statement should be encapsulated as a function -> "isDieVisible" ... especially considering that the same logic is used in 2 places - i.e. for text rendering as well.

Reply:
there is no big overall difference between this function call and no function call
![no function](https://github.com/ni/nimble/assets/33986780/cc1886e0-5d41-40db-96c4-1709d26c9010)
![function](https://github.com/ni/nimble/assets/33986780/c7042211-8871-4d2c-9576-44702c89fd70)